samtranslator/model/sam_resources.py

```diff
@@ -2279,89 +2284,43 @@ def _construct_appsync_schema(
 
         return schema
 
+    def _construct_ddb_datasources(
+        self, ddb_datasources: Dict[str, aws_serverless_graphqlapi.DynamoDBDataSource], api_id: Intrinsicable[str]
+    ) -> List[DataSource]:
+        datasources: List[DataSource] = []
 
-class SamGraphQLDataSource(SamResourceMacro):
-    """SAM GraphQL Data Source Macro (WIP)."""
-
-    # TODO: This currently only supports DataSource with a defined ServiceRoleArn.
-    #       The proposed connector change will come in a follow up PR.
-    resource_type = "AWS::Serverless::GraphQLDataSource"
-    property_types = {
-        "ApiId": PassThroughProperty(True),
-        "Name": Property(False, IS_STR),
-        "Type": Property(True, IS_STR),
-        "Description": PassThroughProperty(False),
-        "ServiceRoleArn": Property(False, IS_STR),
-        "DynamoDBConfig": Property(False, IS_DICT),
-    }
+        for relative_id, ddb_datasource in ddb_datasources.items():
+            cfn_datasource = DataSource(
+                logical_id=relative_id, depends_on=self.depends_on, attributes=self.resource_attributes
+            )
 
-    ApiId: PassThrough
-    Type: str
-    DynamoDBConfig: Optional[Dict[str, Any]]
-    Name: Optional[str]
-    Description: Optional[PassThrough]
-    ServiceRoleArn: Optional[str]
+            cfn_datasource.ServiceRoleArn = ddb_datasource.ServiceRoleArn
+            cfn_datasource.Name = ddb_datasource.Name if ddb_datasource.Name else relative_id
+            cfn_datasource.Type = "AMAZON_DYNAMODB"
+            cfn_datasource.ApiId = api_id
 
-    @cw_timer
-    def to_cloudformation(self, **kwargs: Any) -> List[Resource]:
-        model = self.validate_properties_and_return_model(aws_serverless_graphqldatasource.Properties)
+            if ddb_datasource.Description:
+                cfn_datasource.Description = cast(str, ddb_datasource.Description)
```

Comment:
The model says it's a pass-through, so we don't know that it's a string. It could be a complex intrinsic.